cmake-modules/AzureTransportAdapters.cmake

```diff
@@ -24,14 +28,16 @@ if (WIN32 OR MINGW OR MSYS OR CYGWIN)
     # WinHTTP selected by user on CMake input 
     # OR Nothing selected by CMake input (not libcurl or custom). Then set default for Windows.
     
-    if (NOT BUILD_TRANSPORT_WINHTTP AND NOT BUILD_TRANSPORT_CUSTOM)
+    if (NOT BUILD_TRANSPORT_WINHTTP AND NOT BUILD_TRANSPORT_CUSTOM AND NOT NO_AUTOMATIC_TRANSPORT_BUILD)
```

Comment:
This expression is starting to highlight the logical complexity we are introducing with more build flags.

Instead, I would encourage us to lean towards simplifying this further, to address to the core usage scenario.

Reply:
IMHO, the `'if` expressions in this file are too complicated. I believe this can be achieved in a simpler form. Something like:

```
if (NOT NO_AUTOMATIC_TRANSPORT_BUILD)
if (NOT (BUILD_TRANSPORT_WINHTTP or BUILD_TRANSPORT_CURL or BUILD_TRANSPORT_CUSTOM))
if (WINDOWS)
set(BUILD_TRANSPORT_WINHTTP ON)
else
set (BUILD_TRANSPORT_CURL ON)
endif()
endif() // No Customer Provided Transport
endif() // NO_AUTOMATIC_TRANSPORT_BUILD
if (BUILD_TRANSPORT_WINHTTP)
<winhttp stuff>
endif()
if (BUILD_TRANSPORT_CURL)
<curl stuff>
endif()

Reply:
@LarryOsterman, I wrote a simple condition for `if (NO_AUTOMATIC_TRANSPORT_BUILD)`, and then placed all the existing logic into an `else()`.